src/Nethermind/Nethermind.Synchronization/StateSync/PeerInfoExtensions.cs

```diff
@@ -0,0 +1,28 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Stats.Model;
+using Nethermind.Synchronization.Peers;
+
+namespace Nethermind.Synchronization.StateSync;
+
+public static class PeerInfoExtensions
+{
+    public static bool CanGetNodeData(this PeerInfo peerInfo)
+    {
+        if (!peerInfo.SyncPeer.Node.AgreedCapability.TryGetValue("eth", out int ethVersion))
+        {
+            return false;
+        }
+
+        if (ethVersion <= 66) return true;
+        if (peerInfo.SyncPeer.Node.ClientType == NodeClientType.Nethermind) return true; // Nethermind will still answer to it
```

Comment:
Will it? I think not?

Reply:
It will ignore the message https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V67/Eth67ProtocolHandler.cs#L43

Reply:
On the other hand we shouldn't send `GetNodeData` to `eth67` peer. If we don't have `eth66` or `snap` protocol, we shouldn't send anything: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.Synchronization/StateSync/StateSyncDispatcher.cs#L39-L45

Reply:
Ahh it is changed here too - yeah I don't think this method `CanGetNodeData` is needed - Nethermind will not respond to `GetNodeData` if connected via `eth67` or above, and all old capabilities (`eth65` and below) are deprecated, so we can use `GetNodeData` only with peers connected via `eth66`, so it was fine before (just `peer.Node.EthDetails.Equals("eth66")` in `StateSyncDispatcher`)